src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect } from 'react';
+
+import TodoItem from '../TodoItem/TodoItem';
+
+import { getTodos } from '../../api/todos';
+import { useTodos } from '../../hooks/useTodos';
+import { wait } from '../../utils/fetchClient';
+import { Filter } from '../../types';
+
+export const TodoList: React.FC = () => {
+  const { todos, setTodos, filter, setError, setIsLoading } = useTodos();
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        setIsLoading(true);
+        const fetchedTodos = await getTodos();
+
+        setTodos(fetchedTodos);
+      } catch {
+        setError('Unable to load todos');
+        setIsLoading(false);
+
+        wait(3000).then(() => {
+          setError('');
+        });
```

Comment:
- `setIsLoading(false);` you should remove the loader in both cases, so do it in the finally block.
- Remove errors in 3 seconds you will do it a few more times(when adding/deleting/etc), so create a handler that will do it for you, for example just pass to it an error it will set an error and then remove it in 3 sec.